rotatebykarray.py

```diff
@@ -0,0 +1,11 @@
+#timecomplexity is o(n)
+#space complexity is o(1)
```

Comment:
This solution is equivalent to creating a new array...Under the hood...Python is creating two arrays of length k and length n-k respectively and then merging it